### PR TITLE
CSA-64 Implement /roadmaps/search endpoint

### PR DESCRIPTION
**Pull request by AlfurIvan**

### CODE REVIEW

File: roadmap/admin.py [hunk 1]
```diff
@@ -1,3 +1,31 @@
 from django.contrib import admin
 
-# Register your models here.
+from roadmap import models
```
Comment (ssliadniev) [inline roadmap/admin.py:3]: let's import specific models from `roadmap.models`

File: roadmap/admin.py [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+
+class SmallModelAdmin(admin.ModelAdmin):
+    list_display = ('name', 'id')
+
+
+admin.site.register(models.MetaDomain, SmallModelAdmin)
+admin.site.register(models.LocalDomain, SmallModelAdmin)
+admin.site.register(models.KnowledgeType, SmallModelAdmin)
+admin.site.register(models.EntryLevel, SmallModelAdmin)
+admin.site.register(models.Tag, SmallModelAdmin)
```
Comment (ssliadniev) [inline roadmap/admin.py:31]: Not best approach to create one admin model for several models. 
It reduce code duplication, but it lead to less flexibility and maintainability in the future.
So, let's create a separate admin model for each roadmap related models.

File: roadmap/views.py [hunk 3]
```diff
@@ -1,15 +1,73 @@
...
+    serializer_class = RoadmapSearchSerializer
+    pagination_class = RoadmapSearchPagination
+
+    def get_queryset(self):
+        search_query = self.request.GET.get('search')
+
+        if not search_query:
+            raise NotAcceptable({
+                "message": "Parameter <search> required.",
+                "code": status.HTTP_406_NOT_ACCEPTABLE
```
Comment (bogdan-bridgewise) [inline roadmap/views.py]: why do we return 406 and not 400 ? look what errors could be return in open api schema

**Review by bogdan-bridgewise — CHANGES_REQUESTED** [review]

**Review by bogdan-bridgewise — APPROVED** [review]

File: roadmap/views.py [hunk 4]
```diff
@@ -1,15 +1,75 @@
...
+        try:
+            obj = super().get_object()
+        except Http404 as e:
+            raise NotFound({"message": str(e), "code": 404})
+        else:
+            return obj
+
+
+class DomainListAPIView(generics.ListAPIView):
+    permission_classes = [IsNotDefaultUser]
```
Comment (bogdan-bridgewise) [inline roadmap/views.py]: we should allow anyone to get domains list

File: roadmap/views.py [hunk 5]
```diff
@@ -26,6 +86,7 @@ def list(self, request, *args, **kwargs):
         return Response(data)
 
 
-class EntryLevelListAPIView(ListAPIView):
+class EntryLevelListAPIView(generics.ListAPIView):
+    permission_classes = [IsNotDefaultUser]
```
Comment (bogdan-bridgewise) [inline roadmap/views.py]: same here

**Review by bogdan-bridgewise — DISMISSED** [review]

**Review by ssliadniev — APPROVED** [review]

**Pull request merged**